M1_MAC_SETUP.md

```diff
@@ -0,0 +1,92 @@
+# M1 Mac Setup Guide
+
+This guide should get you set up to run `bigquery-etl` on an M1 or M2 Mac. The
+reason you need to take some extra steps here is because M1/M2 Macs run on
+Apple CPUs that use ARM architecture, as opposed to what older Macs use, Intel
+CPUs on x86 architecture. There are currently some compatibility issues between
+some of our dependencies and ARM. To get around this, we'll use Rosetta, which
+is an x86 emulator.
+
+You're free to set this up however you like, but in this guide we'll set up 2
+parallel Homebrew versions, one for ARM and one for x86 (in Rosetta), create
+aliases for the two, and install all of `bigquery-etl`'s dependencies in
+Rosetta mode.
+
+For tips on maintaining parallel stacks of python and homebrew running with and without Rosetta, see blog posts from [Thinknum](https://medium.com/thinknum/how-to-install-python-under-rosetta-2-f98c0865e012) and [Sixty North](http://sixty-north.com/blog/pyenv-apple-silicon.html).
+
+1. Install Rosetta if you don't have it yet
+
+   ```zsh
+   softwareupdate --install-rosetta
+   ```
+
+2. Set Terminal to run in Rosetta mode:
+   1. In Finder, go to Applications -> Utilities
+   2. Right-click on Terminal and choose Get Info
+   3. Check the box "Open using Rosetta"
```

Comment:
Is great to have a step by step guide. If we have the capacity, we could move all these commands to a makefile. We'd need to maintain it with version changes, but it'd save a lot of time during the onboarding setup, and when upgrading the MAC.